docs/user/troubleshootingbootstrap.md

```diff
@@ -0,0 +1,220 @@
+```
+
+#### Authenticating with bootstrap host for upi
+
+When explicitly using the `gather bootstrap` subcommand, user can either utilize installer's discovery mechanism like detailed [above](#authenticating-with bootstrap host-for-ipi) or you can provide the keys using `--key` flag.
+
+An example of a invocation for a cluster with three control-plane machines would be,
+
+```sh
+openshift-install gather bootstrap --key ${KEY_1} --key ${KEY_2} --bootstrap ${BOOTSTRAP_HOST_IP} --master ${CONTROL_PLANE_1_HOST_IP} --master ${CONTROL_PLANE_2_HOST_IP} --master ${CONTROL_PLANE_3_HOST_IP}
+```
+
+## Understanding the bootstrap failure log bundle
+
+Here's what a log bundle looks like,
+
+```console
+.
+├── bootstrap
+├── control-plane
+├── failed-units.txt
+├── rendered-assets
+├── resources
+└── unit-status
+
+5 directories, 1 file
+```
+
+### file: failed-units.txt
+
+The failed-units.txt contains a list of all the **failed** systemd units on the bootstrap host.
+
+### directory: unit-status
+
+The unit-status directory contains the details of each failed systemd unit from [failed-units][#file-failed-units-txt]
+
+### directory: bootstrap
+
+The bootstrap directory consists of all the important logs and files from the bootstrap host. There are 2 sub directories for the bootstrap host
+
+```console
+bootstrap
+├── containers
+├── journals
+└── pods
+
+3 directories, 0 files
+```
+
+#### directory: bootstrap/containers
+
+The containers directory contains the descriptions and logs from all the containers created by the kubelet using CRIO for the static pods.
+This directory contains all the operators or their operands running on the bootstrap host in special bootstrap modes. For example the machine-config-server container, or the bootstrap-kube-controlplane pods etc.
+
+For each container the directory has two files,
+
+* `<human readable id>.log`, which contains the log of the container.
+* `<human readable id>.inspect`, which containts the information about the container like the image, volume mounts, arguments etc.
+
+#### directory: bootstrap/journals
+
+The journals directory contains the logs for *important* systemd units. These units are,
+
+* `release-image.log`, the release-image unit is responsible for pulling the Release Image to the bootstrap host.
+* `crio-configure.log` and `crio.log`, these units are responsible for configuring the CRI-O on the bootstrap host and CRI-O daemon respectively.
+* `kubelet.log`, the kubelet service is responsible for running the kubelet on the bootstrap host. The kubelet on the bootstrap host is responsible for running the static pods for etcd, bootstrap-kube-controlplane and various other operators in bootstrap mode.
+* `approve-csr.log`, the approve-csr unit is responsible for allowing control-plane machines to join OpenShift cluster. This unit performs the job of in-cluster approver while the bootstrapping is in progress.
+
```

Comment:
nit: your previous list entries had no intervening blank lines; probably drop this one for consistency.